### PR TITLE
Enable llama model training profiler

### PR DESCRIPTION
**Pull request by hongdami**
Still on going...
- 1. Run on cuda without torchrun
- 2. Run cuda DDP: copy the profiler file from utils cause the version is too old to support llama
- 3. Test the NCPL and CUDA Shim (TODO)
- 4. Enable XPU DDP support
- 5. Test the Zeshim and OCPL

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/pytorch/language-modeling/llama.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        train_dataloader = DataLoader(
+            train_dataset,
+            shuffle=True,
+            collate_fn=default_data_collator,
+            batch_size=batch_size,
+            pin_memory=True
+        )
+        eval_dataloader = DataLoader(
+            eval_dataset,
+            shuffle=True,
```
Comment (killight98) [inline examples/pytorch/language-modeling/llama.py]: do not shuffle eval dataset.

File: examples/pytorch/language-modeling/llama.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+            task_type=TaskType.CAUSAL_LM,
+            inference_mode=False,
+            r=8, lora_alpha=16,
+            lora_dropout=0.05,
+            target_modules=["q_proj", "v_proj"],
+        )
+        model = AutoModelForCausalLM.from_pretrained(
+            self.model_name_or_path,
+        )
+        # TODO: Check if half is must
```
Comment (killight98) [inline examples/pytorch/language-modeling/llama.py]: remove this comment

File: examples/pytorch/language-modeling/llama.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+        model = get_peft_model(model, peft_config)
+        model.print_trainable_parameters()
+        logger.info(model)
+        return model
+
+    def train(self):
+        # parameter
+        lr = 3e-3
```
Comment (killight98) [inline examples/pytorch/language-modeling/llama.py]: need this big lr value?
